test/node_roles_functional/test_suites/all/initializer_context.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { PluginFunctionalProviderContext } from '../../services';
+import '@kbn/core-provider-plugin/types';
+
+export default function ({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('initializer context', () => {
+    it('passes node roles to server PluginInitializerContext', async () => {
+      await supertest.get('/core_plugin_initializer_context/node/roles').expect(200, {
+        backgroundTasks: true,
+        ui: true,
+      });
```

Comment:
I would have used an integration test instead of FTR to test that. Would have been easier with less boilerplate for the same value ihmo. But maybe you foresee more tests being added that may not be testable via IT?

Reply:
I had wanted to do integration tests, but there were two reasons I went with functional tests:
1. I thought there might be a chance we want to add more tests over time that would fall outside the scope of what integration tests cover, and figured it'd be easier to have the boilerplate already "done" for this feature.
2. I wanted to test that the whole thing worked end-to-end from configuring `node.roles` to it being injected in `PluginInitializerContext` during plugin discovery. I couldn't think of a way to do this in an integration test, as it really requires a custom test plugin to read the initContext and return the values to test against, which (AFAIK) would require hooking into plugin discovery during the integration test setup, which sorta defeats the purpose of the end-to-end test. But I very well might be overlooking something?

Reply:
> I couldn't think of a way to do this in an integration test, as it really requires a custom test plugin to read the initContext and return the values to test against

Yea, that seems correct.